typing/ctype.ml

```diff
@@ -3233,19 +3239,20 @@ let expand_head_trace env t =
 *)
 
 let filter_arrow env t l =
-  let t = expand_head_trace env t in
```

Comment:
It looks like the `assert false` change could be confined to the definition of `t`.